android/app/src/main/java/com/coboltforge/dontmind/multivnc/AbstractGestureInputHandler.java

```diff
@@ -54,38 +52,25 @@ public boolean onTouchEvent(MotionEvent evt) {
 	 */
 	@Override
 	public boolean onScale(ScaleGestureDetector detector) {
-		boolean consumed = true;
-		//if (detector.)
-		//Log.i(TAG,"Focus("+detector.getFocusX()+","+detector.getFocusY()+") scaleFactor = "+detector.getScaleFactor());
-		// Calculate focus shift
 		float fx = detector.getFocusX();
 		float fy = detector.getFocusY();
-		double xfs = fx - xInitialFocus;
-		double yfs = fy - yInitialFocus;
-		double fs = Math.sqrt(xfs * xfs + yfs * yfs);
-		if (Math.abs(1.0 - detector.getScaleFactor())<0.02)
-			consumed = false;
-		if (fs * 2< Math.abs(detector.getCurrentSpan() - detector.getPreviousSpan()))
```

Comment:
any idea why it was that way @gujjwal00 ?

Reply:
It is handling two separate issues: 
- During scaling, `onScale` callback can be invoked by Android for very small amount of scaling factor. Returning false from these calls will cause them to accumulate. They are handled once they are large enough (± .02). This is what `consumed` is used for. This PR reduces the delta to ± .01
- Variables `inScaling`, `xInitialFocus`, `yInitialFocus`, `fs` etc. and the check on line 68 are used to differentiate between 'Pinch' & 'Two-finger Swipe/Scroll'. For pinch gesture, focus shift (`fs`) would be less then the change in finger span (`detector.getCurrentSpan() - detector.getPreviousSpan()`). This PR removes this logic and lets Android differentiate between these gestures.

I think we can allow smooth scaling even if we want to keep the  2nd behavior. But there will always be edge cases because these two gestures can sometimes be ambiguous.

Reply:
On a second thought, let me look more into this tomorrow.

Reply:
OK. I was asking out of curiosity. If the new code works, I'm fine with it.